### DISCUSSION TITLE
How to migrate raw OSD to PVC based OSD

### DISCUSSION THREAD

**Discussion by henryzhangsta** [rook/rook#8332] [Q&A]
With the merging of https://github.com/rook/rook/pull/7969, there are now docs on how to mount a raw disk as an OSD disk. My question is if I already have disks configured using the old `nodes:` method in the `CephCluster` CRD, is there a way to migrate those OSDs to PVC based without draining and refilling the drives with data?

Does the capacity of the drives in the PVC matter? Do I need to set it equal to the size of the underlying block device?

**Comment by satoru-takeuchi** [comment]
>  is there a way to migrate those OSDs to PVC based without draining and refilling the drives with data?

Unfortunately, there is no such way yet.
